athena-elasticsearch/src/test/java/com/amazonaws/athena/connectors/elasticsearch/integ/ElasticsearchIntegTest.java

```diff
@@ -249,4 +278,48 @@ public void selectColumnWithPredicateIntegTest()
         assertEquals("Wrong number of DB records found.", 1, titles.size());
         assertTrue("Movie title not found: Interstellar.", titles.contains("Interstellar"));
     }
+
+    @Test
+    public void selectFloat8TypeTest()
+    {
+        logger.info("--------------------------------------");
+        logger.info("Executing selectFloat8TypeTest");
+        logger.info("--------------------------------------");
+
+        String query = String.format("select float8_type from %s.%s.%s;",
+                lambdaFunctionName, INTEG_TEST_DATABASE_NAME, TEST_DATATYPES_TABLE_NAME);
+        List<Row> rows = startQueryExecution(query).getResultSet().getRows();
+        if (!rows.isEmpty()) {
+            // Remove the column-header row
+            rows.remove(0);
+        }
+        List<Double> values = new ArrayList<>();
```

Comment:
Similar comment here as earlier about using .map() instead of .forEach() for these kinds of situations.

Reply:
I don't feel strongly as to whether or not you should change this, which is why I approved the pull request in addition to leaving the comments before.

But if you're wondering about the reasoning:

Its a declarative vs imperative style (where declarative is preferred when possible).

More formally this (creating the list and then looping through it to add to it) is an antipattern called ITM (initialize then modify): https://www.youtube.com/watch?v=VV9vwFsaQ6U

In scala a .map() is always used for situations like this instead (just transforming one collection into another) vs a .foreach() which is meant for side effecting operations: https://stackoverflow.com/a/44042773

The youtube link above is mostly about c++ (although he has examples in python as well) and stackoverflow link is on scala, so its not a concept that is isolated to any single language.

It only seems roundabout because its not something that most Java developers are used to (until Java 8 and beyond), but this is also why they added all of this stuff to Java.